src/utils/controller.ts

```diff
@@ -0,0 +1,13 @@
+import type { NextFunction, RequestHandler, Response } from "express";
+
+/**
+ * Turn an async fn whose req is your own typed Request
+ * into a full Express RequestHandler (with proper error bubbling).
+ */
+export function asyncController<TReq>(
```

Comment:
```suggestion
export function asyncController<TReq extends Request>
```

May be a bit more type safer

Reply:
Awesome idea, fixed!